sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java

```diff
@@ -0,0 +1,35 @@
+package com.mesosphere.sdk.dcos.http;
+
+import org.apache.http.impl.client.DefaultRedirectStrategy;
+
+import org.apache.http.annotation.Contract;
+import org.apache.http.annotation.ThreadingBehavior;
+
+@Contract(
+        threading = ThreadingBehavior.IMMUTABLE
+)
+/**
+ * A copy of {@link org.apache.http.impl.client.LaxRedirectStrategy} with addition support for PUT method redirect
+ * that is necessary for SecretsClient.
+ */
+public class CustomRedirectStrategy extends DefaultRedirectStrategy {
```

Comment:
That would keep things simpler.

@mhrabovcin can you make that change and add a unit test for the isRedirectable method?